airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -408,7 +408,7 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Ite
                 # avoid checking `deleted` records for substreams, a.k.a `Metafields` streams,
                 # since `deleted` records are not available, thus we avoid HTTP-400 errors.
                 if self.deleted_cursor_field not in record:
-                    yield {self.slice_key: record[self.nested_record]}
+                    yield {self.slice_key: record[self.nested_record]} | (stream_state or {})
```

Comment:
Can you explain the reason of returning stream_state here?

Reply:
I required access to the original `updated_at` date (from the state file passed in as an argument to the sync execution) in the `request_params` method. I couldn't find a better approach: 

1. The `stream_state` in the `request_params` seems to be always an empty dict
2. `self.get_updated_state` does no good here as we're looking for the value to use for the `updated_at_min` param
3. This connector does not have a `self.state` property

I'm eager to know of a better way of achieving this as it felt a bit hacky to me too. Thanks @marcosmarxm 🙌